tox.ini

```diff
@@ -1,48 +1,94 @@
-[tox]
-envlist= py38,style,bandit
-
-[testenv]
-extras= test
+[testtenv]
+extras = test
 # astropy will complain if the home directory is missing
-passenv= HOME
-commands=
-    pytest
-
-[testenv:egg_info]
-deps=
-commands=
-    python setup.py egg_info
-
-[testenv:twine]
-deps=
-    twine
-commands=
-    twine check {distdir}/*
+passenv =
+    HOME
+deps =
+    pytest-xdist
+commands =
+    pip freeze
+    pytest -n auto
 
 [testenv:style]
-deps=
+deps =
     flake8
-commands=
+commands =
     flake8 --count src
 
-[testenv:coverage]
-deps=
-    codecov
-    coverage
-commands=
-    coverage run --source=stpipe -m pytest
-    coverage report -m
-    codecov -e TOXENV
-passenv= TOXENV CI TRAVIS TRAVIS_* CODECOV_* DISPLAY HOME
-
 [testenv:bandit]
-deps=
+deps =
     bandit
-commands=
+commands =
     bandit -r -ll src
 
 [testenv:build-docs]
-basepython= python3.8
-extras= docs
-commands=
+basepython = python3.8
+extras = docs
+commands =
     sphinx-build -W docs/source build/docs
+
+[testenv:coverage]
+basepython = python3.9
+extras = test
+deps =
+    codecov
+    pytest-cov
+    pytest-xdist
+passenv =
+    CI
+    CODECOV_*
+    CRDS_*
+    DISPLAY
+    HOME
+    TOXENV
+    TRAVIS
+    TRAVIS_*
+commands =
+    pip freeze
+    pytest -n auto --cov-report=xml --cov={toxinidir}/src/stpipe
+    coverage report -m
+
+[testenv:jwst-cov]
+basepython = python3.9
+extras = test
+deps =
+    codecov
+    jwst[test] @ git+https://github.com/spacetelescope/jwst.git
+    pytest-cov
+    pytest-xdist
+passenv =
+    CI
+    CODECOV_*
+    CRDS_*
+    DISPLAY
+    HOME
+    TOXENV
+    TRAVIS
+    TRAVIS_*
+commands =
+    pip freeze
+    pytest -n auto --cov-report=xml --cov={toxinidir}/src/stpipe --ignore-glob=timeconversion --ignore-glob=associations --pyargs {posargs:jwst}
+    coverage report -m
+
+[testenv:romancal-cov]
+basepython = python3.9
+extras = test
+deps =
+    codecov
+    pytest-cov
+    pytest-xdist
+    romancal[test] @ git+https://github.com/spacetelescope/romancal.git
+passenv =
+    CI
+    CODECOV_*
+    CRDS_*
+    DISPLAY
+    HOME
+    TOXENV
+    TRAVIS
+    TRAVIS_*
+commands =
+    pip install -U stpipe==0.3.1
```

Comment:
explicitly install `stpipe==0.3.1` (works for now, but does not test the current commit)